SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/SpatialTestRemotePossession.cpp

```diff
@@ -2,22 +2,35 @@
 
 #include "SpatialTestRemotePossession.h"
 
+#include "CrossServerPossessionGameMode.h"
 #include "Kismet/GameplayStatics.h"
 #include "Net/UnrealNetwork.h"
 #include "SpatialFunctionalTestFlowController.h"
 #include "SpatialGDKFunctionalTests/SpatialGDK/TestActors/TestPossessionPawn.h"
 #include "TestPossessionPlayerController.h"
+#include "TestWorkerSettings.h"
 
 const float ASpatialTestRemotePossession::MaxWaitTime = 2.0f;
 
 ASpatialTestRemotePossession::ASpatialTestRemotePossession()
-	: Super()
+	: ASpatialFunctionalTest(EMapCategory::CI_NIGHTLY_SPATIAL_ONLY, 1)
 	, LocationOfPawn(500.0f, 500.0f, 50.0f)
 {
 	Author = "Jay";
 	Description = TEXT("Test Actor Remote Possession");
 }
 
+void ASpatialTestRemotePossession::CreateCustomContentForMap()
+{
+	GetWorldSettings()->SetMultiWorkerSettingsClass(UTest2x2FullInterestWorkerSettings::StaticClass());
+	GetWorldSettings()->DefaultGameMode = ACrossServerPossessionGameMode::StaticClass();
+
+	if (this->GetClass() == ASpatialTestRemotePossession::StaticClass())
```

Comment:
I think this is a bit of a weird design to deal with inherited classes not being standalone. It hides properties of the derived classes in the base class in a way I would personally easily miss.
I suggest the following.
Remove: `SetIsNotStandaloneTest()`
Make: `bIsStandaloneTest` -> `virtual bool IsStandaloneTest() { return false; }` in `ASpatialFunctionalTest` and override with returning true / false as appropriate in derived classes.
TBH I don't really like ASpatialTestRemotePossession is both a base class for tests and a test in itself, its a code smell to me, which makes this more complicated than it needs to be (you need to override on each of the `ASpatialTestRemotePossession` derived classes `IsStandaloneTest` to return false bc the baseclass is standalone) but addressing that is out of scope.

Reply:
Admit to finding it confusing also.

Reply:
Have changed to overriding ShouldGenerateMap, I agree it seems slightly more readable

Reply:
Nvm, just learnt more about ue4 abstract classes and realised I can check the UClass for CLASS_Abstract, so doing that instead!